### PR TITLE
Make changes to the FCF technote

### PR DESCRIPTION
**Pull request by dlongnecke-cray**
This PR makes changes to the FCF technote to reflect changes made in
the 1.31 release. It also adjusts the terminology of the technote to
use the term "first-class procedure" instead of "first-class function"
throughout.

Reviewed by @vasslitvinov. Thanks!

### CODE REVIEW

File: doc/rst/technotes/firstClassProcedures.rst [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+assigned to a procedure defined with different formal names. In the
+below example, assignment of two procedure values will fail because
+the formal names are different:
+
+.. code-block:: chapel
+
+  proc foo(x: int, y: int): int { return x + y; }
+  proc bar(a: int, b: int): int { return a + b; }
+  writeln(foo.type:string);   // 'proc(x: int, y: int): int'
+  var x = foo;                // ^ (same type) ^
```
Comment (vasslitvinov) [inline doc/rst/technotes/firstClassProcedures.rst:62]:
```
// x has the same type as foo: `proc(x: int, y: int): int`
```

File: doc/rst/technotes/firstClassProcedures.rst [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+  writeln(bar.type:string);   // 'proc(a: int, b: int): int'
+  x = bar;                    // Error!
+
+There are scenarios where a user may want to pass around procedures 
+without regard for their formal names. Anonymous ``_`` formals are a
+feature under active development that are intended to support this
```
Comment (vasslitvinov) [inline doc/rst/technotes/firstClassProcedures.rst:68]: that are --> that is ?

File: doc/rst/technotes/firstClassProcedures.rst [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+behavior.
+
+New Syntax for Constructing Anonymous Procedures 
+------------------------------------------------
+
+A new syntax for constructing anonymous procedures has been introduced
+which more closely mirrors traditional procedure definition.
```
Comment (vasslitvinov) [inline doc/rst/technotes/firstClassProcedures.rst:75]: Again, I suggest removing the above sentence "a new syntax for ..." and switch the section name to just "Anonymous Procedures."

File: doc/rst/technotes/firstClassProcedures.rst [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+.. code-block:: chapel
+
+  // Define a procedure named 'foo'.
+  proc foo(x: int, y: int): int { return x + y; }
```
Comment (vasslitvinov) [inline doc/rst/technotes/firstClassProcedures.rst:80]: How about:
// 'foo' is a "named" procedure, not an anonymous procedure

File: doc/rst/technotes/firstClassProcedures.rst [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+  // Define an anonymous procedure bound to the constant variable 'bar'.
+  const bar = proc(a: int, b: int): int { return a + b; };
+
+The Formals of Procedure Types May Be Anonymous
+-----------------------------------------------
+
+.. note::
+
+  This section describes unstable functionality that may not be implemented.
```
Comment (vasslitvinov) [inline doc/rst/technotes/firstClassProcedures.rst:90]: I suggest removing this 'note' in favor of the note below.

File: doc/rst/technotes/firstClassProcedures.rst [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+In the above example, two procedures are declared with different formal
+names, but otherwise identical types. A local variable `x` is declared
+with a procedure type that has anonymous formals. This enables the
+procedures `foo` and `bar` to be assigned freely to `x`.
+
+.. note::
+
+  Currently, it is not possible for procedure definitions to declare
+  anonymous formals. It has been indicated that such a feature might
+  be useful, so it may be added in the future.
```
Comment (vasslitvinov) [inline doc/rst/technotes/firstClassProcedures.rst:123]: How about this:

Currently anonymous formals are not implemented.
This feature may be added in the future.

File: doc/rst/technotes/firstClassProcedures.rst [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+Anonymous procedures may be captured as though they are named procedures,
+by substituting them in places where a procedure name may also appear:
+
+.. code-block:: chapel
+
+  const p = proc(x: int) { return x + 1; };
+  writeln(p(3));  // outputs: 4
+
+Only procedures (defined with the ``proc`` keyword) may be captured.
```
Comment (vasslitvinov) [inline doc/rst/technotes/firstClassProcedures.rst:154]: I suggest adding:

Currently iterators cannot be captured.

**Review by vasslitvinov — APPROVED** [review]
Please see my suggestions

**Comment by dlongnecke-cray** [comment]
Thank you Vass, I believe I've responded to all your feedback. I also took the liberty of rewording the semantics for anonymous formals in procedure types to improve clarity.

**Pull request merged**